docs/dev/README.md

```diff
@@ -0,0 +1,218 @@
+
+## Using an IDE
+### Using Visual Studio Code (VSCode)
+
+1)  Install the following extensions:
+    *   Python (Microsoft)
+    *   C/C++ (Microsoft)
+    *   Code Spell Checker (Street Side Software) - optional
+
+2)  Open the `aws-crt-python/` folder.
+
+3)  Edit workspace settings: `cmd+shift+P -> Preferences: Open Workspace Settings`
+    *   Python: Default Interpreter Path - ("python.defaultInterpreterPath" in json view)
+        set the absolute path to Python in your virtual environment. For example: `/Users/janedoe/dev/aws-crt-python/.venv/bin/python`
+        *   Note that the VSCode terminal ignores this setting and will not use your virtual environment by default.
+            You must manually run `source .venv/bin/activate` each time you start using the terminal.
+            Or use the command `cmd+shift+P -> Python: Create Terminal`.
+    *   C_Cpp > Default: Include Path - ("C_Cpp.default.includePath" in json view)
+        *   Add item - set path to Python's C headers.
+            For example: `/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10`
+        *   This is optional, it helps IntelliSense when viewing C files.
+
+4)  Add helpful tasks you can run via `cmd+shift+P -> Tasks: Run Task`
+    *   Copy [this file](vscode/launch.json) to `aws-crt-python/.vscode/tasks.json` for the following tasks:
+        * install - pip install in develop mode. `cmd+shift+B` is a special shortcut for this task
+        * format-python - format all .py files
+
+#### Debugging Python with VSCode
+The VSCode `Testing` tab (lab flask/beaker icon) helps run and debug Python tests.
+From this tab, click Configure Python Tests:
+*   Select a test framework/tool to enable - unittest
+*   Select the directory containing the tests - test
+*   Select the pattern to identify test files - test_*.py
+
+Run tests by mousing over the name and clicking the PLAY button,
+or use the DEBUG button to hit breakpoints in Python code, and see output in the DEBUG CONSOLE.
+
+Note that many tests are skipped unless [specific environment variables](#environment-variables-for-tests) are set.
+You can set these in a `aws-crt-python/.env` file (don't worry it's ignored by git. For example:
+```
+AWS_TEST_IOT_MQTT_ENDPOINT=xxxxxxxxx-ats.iot.xxxxx.amazonaws.com
+AWS_TEST_TLS_CERT_PATH=/Users/janedoe/iot/xxxxx-certificate.pem.crt
+AWS_TEST_TLS_KEY_PATH=/Users/janedoe/iot/xxxxx-private.pem.key
+```
+
+#### Debugging C with VSCode
+Unfortunately, we haven't figured out how to do interactive debugging of the C code.
+Python ultimately builds and links the module together, and it seems to always strip out the debug info.
+Please update this guide if you know how. For now, `printf()` is your best option.
+
+If you suspect the bug is in the external C code (i.e. [aws-c-http](https://github.com/awslabs/aws-c-http))
+and need to do interactive debugging, your best bet is cloning that external project,
+build it in Debug, and step through its tests.
+
+# TODO
```

Comment:
A thought: Should we encourage contributors to also test to make sure the SDK builds/compiles? Like maybe for them to quickly run the pubsub just to make sure nothing there breaks?

Reply:
nope
that is too much hassle
if we want something like that, we should build it into the CI

Seriously, it **is a good idea** to test whether it will break key downstream dependencies. But it is too much hassle to ask first time contributors to a code base to ALSO figure out how to build another package and test it, especially when you can't test the SDK without an IoT Core account, and not everyone contributing to aws-crt-python is doing IoT